apollo-router/src/apollo_studio_interop/mod.rs

```diff
@@ -19,12 +21,131 @@ use apollo_compiler::executable::InlineFragment;
+            referenced_input_fields: HashMap::new(),
+            referenced_enums: HashMap::new(),
+        }
+    }
+}
+
+/// The aggregation of ExtendedReferenceStats across a number of executions.
+#[derive(Clone, Default, Debug, Serialize)]
+pub(crate) struct AggregatedExtendedReferenceStats {
+    /// A map of parent type to a map of field name to aggregated stats
+    pub(crate) referenced_input_fields:
+        HashMap<String, HashMap<String, AggregatedInputObjectFieldStats>>,
+    /// A map of enum name to a map of enum values to the number of executions that referenced them.
+    pub(crate) referenced_enums: HashMap<String, HashMap<String, u64>>,
+}
+
+impl AggregatedExtendedReferenceStats {
+    pub(crate) fn is_empty(&self) -> bool {
+        self.referenced_input_fields.is_empty() && self.referenced_enums.is_empty()
+    }
+}
+
+impl AddAssign<ExtendedReferenceStats> for AggregatedExtendedReferenceStats {
+    fn add_assign(&mut self, other_stats: ExtendedReferenceStats) {
+        // Merge input object references
+        for (type_name, type_stats) in other_stats.referenced_input_fields.iter() {
+            for (field_name, field_stats) in type_stats.iter() {
+                match self
+                    .referenced_input_fields
+                    .entry(type_name.to_string())
+                    .or_default()
+                    .entry(field_name.to_string())
+                {
+                    Entry::Occupied(mut entry) => {
+                        let stats: &mut AggregatedInputObjectFieldStats = entry.get_mut();
+                        stats.referenced += if field_stats.referenced { 1 } else { 0 };
+                        stats.null_reference += if field_stats.null_reference { 1 } else { 0 };
+                        stats.undefined_reference += if field_stats.undefined_reference {
+                            1
+                        } else {
+                            0
+                        };
+                    }
+                    Entry::Vacant(entry) => {
+                        entry.insert(AggregatedInputObjectFieldStats {
+                            referenced: if field_stats.referenced { 1 } else { 0 },
+                            null_reference: if field_stats.null_reference { 1 } else { 0 },
+                            undefined_reference: if field_stats.undefined_reference {
+                                1
+                            } else {
+                                0
+                            },
+                        });
+                    }
+                }
+            }
+        }
+
+        // Merge enum references
+        for (enum_name, enum_values) in other_stats.referenced_enums.iter() {
+            for enum_value in enum_values.iter() {
+                match self
+                    .referenced_enums
+                    .entry(enum_name.to_string())
+                    .or_default()
+                    .entry(enum_value.to_string())
```

Comment:
if we're in a hot path here. We already spotted that using the entry api might be less performant